pkg/gadgets/helpers.go

```diff
@@ -224,3 +225,57 @@ func FixBpfKtimeGetBootNs(programSpecs map[string]*ebpf.ProgramSpec) {
 		removeBpfKtimeGetBootNs(s)
 	}
 }
+
+// OptsWithMountnsMap adds an ebpf.Map to the ebpf.CollectionOptions and creates
+// the ebpf.CollectionOptions if it does not exist
+func OptsWithMountnsMap(opts *ebpf.CollectionOptions, mountnsMap *ebpf.Map) *ebpf.CollectionOptions {
+	if opts == nil {
+		opts = &ebpf.CollectionOptions{}
+	}
+	if opts.MapReplacements == nil {
+		opts.MapReplacements = map[string]*ebpf.Map{}
+	}
+	if mountnsMap != nil {
+		opts.MapReplacements["mount_ns_filter"] = mountnsMap
+	}
+
+	return opts
+}
+
+// LoadeBPFSpec collects operations that are duplicated across gadgets.
+// It replaces filter map and calls the necessary function to load
+// Maps and Programs into the kernel
+func LoadeBPFSpec(
+	mountnsMap *ebpf.Map,
+	spec *ebpf.CollectionSpec,
+	consts map[string]interface{},
+	objs interface{},
+) error {
+	FixBpfKtimeGetBootNs(spec.Programs)
+
+	mapReplacements := map[string]*ebpf.Map{}
+	filterByMntNs := false
+
+	if mountnsMap != nil {
+		filterByMntNs = true
+		mapReplacements["mount_ns_filter"] = mountnsMap
+	}
+
+	if consts == nil {
+		consts = map[string]interface{}{}
+	}
+
+	consts["filter_by_mnt_ns"] = filterByMntNs
+
+	if err := spec.RewriteConstants(consts); err != nil {
+		return fmt.Errorf("error RewriteConstants: %w", err)
```

Comment:
nit: about adding `error`, `fail`, etc in these messages -> https://github.com/inspektor-gadget/inspektor-gadget/issues/881

(I know we have a bunch of code using that, but we're trying not to introduce new code with that issue)

Reply:
Good to know about this effort. Re-worded that ✅